### PR TITLE
fix: fixes for latest djs update

### PR DESCRIPTION
**Pull request by favna**

### CODE REVIEW

File: src/listeners/twitch/twitchStreamOffline.ts [hunk 1]
```diff
@@ -43,7 +43,7 @@ export class UserListener extends Listener<Events.TwitchStreamOffline> {
 
 				// Retrieve the channel to send the message to
 				const channel = guild.channels.cache.get(guildSubscription.channelId) as TextBasedChannelTypes;
-				if (isNullish(channel) || !canSendMessages(channel)) {
+				if (isNullish(channel) || !canSendMessages(channel) || channel.isDMBased()) {
```
Comment (kyranet) [inline src/listeners/twitch/twitchStreamOffline.ts:46]: This should be casted to a guild text based channel instead.

File: tsconfig.base.json [hunk 2]
```diff
@@ -1,6 +1,7 @@
 {
 	"extends": ["@sapphire/ts-config", "@sapphire/ts-config/extra-strict", "@sapphire/ts-config/decorators", "@sapphire/ts-config/verbatim"],
 	"compilerOptions": {
-		"removeComments": true
+		"removeComments": true,
+		"skipLibCheck": true
```
Comment (kyranet) [inline tsconfig.base.json:5]: Any reason for this change?

File: package.json [hunk 3]
```diff
@@ -114,13 +114,12 @@
...
 		"tsc-watch": "^6.2.0",
-		"typescript": "^5.4.5",
+		"typescript": "~5.4.5",
 		"vitest": "^2.1.3"
 	},
 	"resolutions": {
-		"@types/node": "20.11.5",
 		"ansi-regex": "^5.0.1",
-		"discord-api-types": "^0.37.98",
+		"discord-api-types": "0.37.97",
```
Comment (kyranet) [inline package.json:122]: I'm pretty sure latest djs is on latest -types (and with ^), unless that's still not released cc: @vladfrangu

**Review by kyranet — CHANGES_REQUESTED** [review]

File: tsconfig.base.json (hunk 2 above)
Comment (favna) [reply tsconfig.base.json:5]: error in /decorators where MethodDecorator is somehow no longer generic

File: package.json (hunk 3 above)
Comment (favna) [reply package.json:122]: I checked what's in yarn.lock... :\

**Comment by kyranet** [comment]
Superseded by #2668 with the fixes I wanted to have

**Pull request closed without merging**